src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -240,15 +234,28 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
                 }
 
                 IEnumerable<RestoreTargetGraph> graphs = null;
-                using (var restoreGraphTelemetry = TelemetryActivity.CreateTelemetryActivityWithNewOperationIdAndEvent(parentId: _operationId, eventName: GenerateRestoreGraph))
+                if (_success)
```

Comment:
If _success was false from the beginning why it was needed to do the lock and noop evaluation and not fail directly?

Reply:
Initially I was hoping no-op would still work, so we could avoid having to write the assets file more than once, until the nomination error was fixed. Since I couldn't get that to work, I guess it's pointless.

Reply:
I looked into this further, and there are several variables that are needed at the end of the method, during asset file and DGSpec writing. I don't think that it would be worthwhile to refactor the code to skip the no-op check, but still get the values for these variables.